analyzers/src/SonarAnalyzer.CSharp/Rules/MethodsShouldNotHaveIdenticalImplementations.cs

```diff
@@ -47,5 +58,11 @@ public sealed class MethodsShouldNotHaveIdenticalImplementations : MethodsShould
             && firstMethod.Body.IsEquivalentTo(secondMethod.Body, false);
 
         protected override SyntaxToken GetMethodIdentifier(IMethodDeclaration method) => method.Identifier;
+        protected override bool IsExcludedFromBeingExamined(ISymbol nodeContainingSymbol) =>
+            nodeContainingSymbol.Kind != SymbolKind.NamedType && !(nodeContainingSymbol is INamespaceSymbol namespaceSymbol
+                                                                   && namespaceSymbol.IsGlobalNamespace
```

Comment:
Can't we just provide the node AND containing symbol? And then if the node is CompilationUnit - just check that this is toplevelprogram? I think that this would make this check more explainable.

Reply:
I tried to apply this but didn't manage to write something that looks good.
Any hints?
Meanwhile, I changed this function to evaluate the opposite (since I think it makes it more readable), and broke it into two pieces for the CSharp version for the same reason.
WDYT?

Reply:
See below. Moreover, I think that our code style is more friendly to positive conditions, rather negative. So I think that the way it was before - matched this style better.